source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -1,33 +1,98 @@
+using System;
+using System.Collections.Generic;
 using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.Mineplacer;
 
 namespace F0.Minesweeper.Logic
 {
 	internal sealed class Minefield : IMinefield
 	{
 		private readonly uint width, height, mineCount;
+		private IMineplacer mineplacer;
+		private bool isFirstUncover = true;
+		private bool[,] minefield;
+		private string DebugVisualization
```

Comment:
I'd prefer `DebuggerDisplayAttribute` here:
see https://docs.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute
_Visual Studio_ has a code fix for that: https://docs.microsoft.com/en-us/visualstudio/ide/reference/add-debugger-display-attribute